test/functional/ui/screen.lua

```diff
@@ -450,8 +450,14 @@ end
 function Screen:expect_unchanged(waittime_ms, ignore_attrs, request_cb)
   waittime_ms = waittime_ms and waittime_ms or 100
   -- Collect the current screen state.
-  self:sleep(waittime_ms, request_cb)
-  local kwargs = self:get_snapshot(nil, ignore_attrs)
+  local kwargs
+  local function notification_cb(method, args)
+    assert(method == 'redraw')
+    assert(self:_redraw(args))  -- did flush
+    kwargs = self:get_snapshot(nil, ignore_attrs)
+  end
+  run_session(self._session, request_cb, notification_cb, nil, 0)
```

Comment:
I am not sure if this (whole new block) is better than `self:sleep(0, request_cb)`?

Reply:
Right

Reply:
@justinmk I understand that as if you are also wondering?

@bfredl can you shed some (more) light, please?